toolkit/tools/pkg/imagecustomizerlib/customizeboot.go

```diff
@@ -57,10 +60,179 @@ func handleKernelCommandLine(extraCommandLine string, imageChroot *safechroot.Ch
 	newGrub2ConfigFile := grub2ConfigFile[:insertIndex] + extraCommandLine + " " + grub2ConfigFile[insertIndex:]
 
 	// Update grub.cfg file.
-	err = os.WriteFile(grub2ConfigFilePath, []byte(newGrub2ConfigFile), 0)
+	err = writeGrub2ConfigFile(newGrub2ConfigFile, imageChroot)
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Updates the kernel command-line args with the new SELinux mode.
+//
+// See, installutils.setGrubCfgSELinux()
+func updateSELinuxCommandLine(selinuxMode imagecustomizerapi.SELinux, imageChroot *safechroot.Chroot) error {
+	logger.Log.Infof("Updating SELinux kernel command-line args")
+
+	grub2Config, err := readGrub2ConfigFile(imageChroot)
+	if err != nil {
+		return err
+	}
+
+	// Try to find any existing SELinux args.
+	selinuxArgsMatch := linuxCommandLineSELinuxRegex.FindStringSubmatchIndex(grub2Config)
+
+	start := 0
+	end := 0
+
+	if selinuxArgsMatch == nil {
+		// SELinux args are not in the grub.cfg file.
+		if selinuxMode == imagecustomizerapi.SELinuxDisabled {
+			// Nothing to do.
+			return nil
+		}
+
+		insertLocationMatch := linuxCommandLineInsertRegex.FindStringIndex(grub2Config)
+		if insertLocationMatch == nil {
+			return fmt.Errorf("failed to find location to insert SELinux kernel command-line args")
```

Comment:
would it be helpful if we print out the regex here for the user? or an explanation of what we're looking for?
I'm thinking of a case where the user has some grub.cfg, and it fails and there is not information to fix the problem.

Reply:
I'm kind of hoping that the error never triggers for a user. But on the off chance it does, then the unique error message ensures that we can quickly diagnose the problem.